packages/tse-backend/src/common/utils/trading-strategy.utils.ts

```diff
@@ -62,3 +68,122 @@ export const isExchangeSupported = (
 ) => {
   return supported.includes(exchangeName);
 };
+
+type PriceSourceFunction = (orderBook: any, ticker: any) => number;
+
+const priceSourceFunctions: Record<PriceSourceType, PriceSourceFunction> = {
+  [PriceSourceType.MID_PRICE]: (orderBook) =>
+    (orderBook.bids[0][0] + orderBook.asks[0][0]) / 2,
+  [PriceSourceType.BEST_ASK]: (orderBook) => orderBook.asks[0][0],
+  [PriceSourceType.BEST_BID]: (orderBook) => orderBook.bids[0][0],
+  [PriceSourceType.LAST_PRICE]: (_, ticker) => ticker.last,
+};
+
+export const getPriceSource = async (
+  exchange: any,
+  pair: string,
+  priceSourceType: PriceSourceType,
+) => {
+  const orderBook = await exchange.fetchOrderBook(pair);
+  const ticker =
+    priceSourceType === PriceSourceType.LAST_PRICE
+      ? await exchange.fetchTicker(pair)
+      : null;
+
+  const priceSourceFunction = priceSourceFunctions[priceSourceType];
+  if (!priceSourceFunction) {
+    throw new Error(`Invalid price source type: ${priceSourceType}`);
+  }
+
+  return priceSourceFunction(orderBook, ticker);
+};
+
+export function adjustOrderAmount(
+  initialOrderAmount: number,
+  layer: number,
+  amountChangeType: AmountChangeType,
+  amountChangePerLayer: number,
+): number {
+  if (layer > 1) {
+    if (amountChangeType === AmountChangeType.FIXED) {
+      return initialOrderAmount + (layer - 1) * amountChangePerLayer;
+    } else if (amountChangeType === AmountChangeType.PERCENTAGE) {
+      let adjustedAmount = initialOrderAmount;
+      for (let i = 1; i < layer; i++) {
+        adjustedAmount += adjustedAmount * (amountChangePerLayer / 100);
+      }
+      return adjustedAmount;
+    }
+  }
+  return initialOrderAmount;
```

Comment:
Simplify that, we don't do nested conditions

Reply:
Fixed: 96d2df91